python/cudf/cudf/utils/dtypes.py

```diff
@@ -425,11 +477,65 @@ def dtype_to_pylibcudf_type(dtype) -> plc.DataType:
     # libcudf types don't support timezones so convert to the base type
     elif isinstance(dtype, pd.DatetimeTZDtype):
         dtype = _get_base_dtype(dtype)
+    elif isinstance(dtype, pd.StringDtype):
+        dtype = CUDF_STRING_DTYPE
     else:
-        dtype = np.dtype(dtype)
+        dtype = pandas_dtypes_to_np_dtypes.get(dtype, dtype)
+        try:
+            dtype = np.dtype(dtype)
+        except TypeError:
+            dtype = cudf.dtype(dtype)
+    # print(type(dtype), dtype)
     return plc.DataType(SUPPORTED_NUMPY_TO_PYLIBCUDF_TYPES[dtype])
 
 
+def dtype_to_pyarrow_type(dtype) -> pa.DataType:
+    if isinstance(dtype, pd.ArrowDtype):
+        return dtype
+    if isinstance(
+        dtype,
+        (cudf.ListDtype, cudf.StructDtype, cudf.core.dtypes.DecimalDtype),
+    ):
+        return dtype.to_arrow()
+    # libcudf types don't support timezones so convert to the base type
+    elif isinstance(dtype, pd.DatetimeTZDtype):
+        dtype = _get_base_dtype(dtype)
+    else:
+        dtype = pandas_dtypes_to_np_dtypes.get(dtype, dtype)
+        try:
+            dtype = np.dtype(dtype)
+        except TypeError:
+            dtype = cudf.dtype(dtype)
+    return pa.from_numpy_dtype(dtype)
+
+
+def dtype_to_pandas_nullable_extension_type(dtype) -> DtypeObj:
+    if isinstance(dtype, pd.ArrowDtype):
+        return dtype_to_pandas_nullable_extension_type(
+            pyarrow_dtype_to_cudf_dtype(dtype)
+        )
+    else:
+        return np_dtypes_to_pandas_dtypes.get(dtype, dtype)
+
+
+def get_dtype_of_same_kind(source_dtype: DtypeObj, target_dtype: DtypeObj):
+    """
+    Given a dtype, return a dtype of the same kind
+    with the same size as the input dtype.
```

Comment:
What does "with the same size` refer to here?

Reply:
Ignore the confusing text. Removed it.